design/one-pager-resource-status.md

```diff
@@ -0,0 +1,135 @@
+  > _(Essentially a default case in a switch block)_
+
+## Design
+
+In contrast to the design [here](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-stack-status-spec.md),
+this design is meant to handle combinations of arbitrary statuses to determine a high level "overall" status.
+
+The following examples are meant to be added to a `ui-schema.yaml` file. The `resourceStatus` key should be added as a root key in that file.
+
+<a name="example-1"></a>
+
+```yaml
+# Example 1
+resourceStatus:
+  paths: .status.conditionedStatus.conditions
+  states:
+    Online:
+    - Ready: 'True'
+      Synchronized: 'True'
+    Offline:
+    - Ready: 'False'
+      Synchronized: 'False'
+    Warning:
+    - Ready: 'False'
+      Synchronized: 'True'
+      SyncFailure: 'True'
+```
+
+<a name="example-2"></a>
+
+```yaml
+# Example 2
+resourceStatus:
+  paths:
+    Ready: .status.conditionedStatus.conditions[?(@.type=='Ready')]
+    Synchronized: .status.conditionedStatus.conditions[?(@.type=='Synchronized')]
+    SyncFailure: # Example 2a
+      status: .status.someRandomStatuses.SyncFailure.value
+      message: .status.someRandomStatuses.SyncFailure.text
+      lastTransitionTime: .status.someRandomStatuses.SyncFailure.updatedTime
+  states:
+    Online:
+    - Ready: True
+      Synchronized: True
+      SyncFailure: False
+    - Ready: True
+      Synchronized: True
+      SyncFailure: not 'True'
+    Offline:
+    - Ready: False
+      Synchronized: False
+    - Ready: True
+      Synchronized: not 'True'
+      SyncFailure: True
+    Warning:
+    - Ready: 'False'
+      Synchronized: 'True'
+      SyncFailure: 'True'
+```
+
+There are two main keys within `resourceStatus`. The first is `paths` which can be a string or map of key/value pairs. Additionally the key/value pairs can be set as <string>: <string> **OR** <string>: <object>.
+
+To expand on how to set the `paths` key a bit:
+
+- If the **states** are using status types that can all be found within a single conditions array that follow the object
+structure of a standard [PodCondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#podcondition-v1-core),
+then you can just set the `paths` key to the JSONPath of that conditions array. (_[Example 1](#example-1)_)
+
+- If the states are using status types from various or arbitrary locations, then you can individually set each path as a
+**key**/**value** pair. (_[Example 2](#example-2)_)
+  > \<status type> / \<JSONPath to condition object>
+
+- Lastly, if you want to be very explicit about the individual values of a particular status type, you can set JSONPaths
+for each individual condition key. (_[Example 2a](#example-2)_)
+
+- As a rule of thumb, you will need to define the `status`, `message` and `lastTransitionTime` for a path key, at the
+bare minimum.
+
+The `states` key should follow these rules:
+
+- There are 3 nested keys which you can define:
```

Comment:
since this is a general UI schema, do you think we should allow arbitrary states to be defined here too? if a UI doesn't know what a state means, it could always ignore it.

Reply:
I will be updating the language to address this